extension/src/popup/components/sendPayment/SendAmount/index.tsx

```diff
@@ -579,9 +579,13 @@ export const SendAmount = ({
             <EditMemo
               memo={transactionData.memo || ""}
               onClose={() => setIsEditingMemo(false)}
-              onSubmit={({ memo }: { memo: string }) => {
+              onSubmit={async ({ memo }: { memo: string }) => {
                 dispatch(saveMemo(memo));
                 setIsEditingMemo(false);
+                // Regenerate transaction XDR with new memo (now reads memo from Redux state inside fetchData)
+                await fetchSimulationData();
+                // Reopen review sheet after memo is saved and XDR is regenerated
+                setIsReviewingTx(true);
```

Comment:
Hm, why automatically open this review sheet here? As a user, I would expect this to come up only after I click `Review Send`. It's also a bit jarring because editing the fee doesn't open the review sheet - it's just editing the memo.

I think we should maintain the original UX of the user needing to click `Review Send` to get to the review sheet